packages/app/lib/models/base.dart

```diff
@@ -59,3 +63,46 @@ abstract class Paginator<T> {
     return next;
   }
 }
+
+Future<List<Map<String, dynamic>>> paginateOverEverything(
```

Comment:
The `FetchAll` pagination list widget contains similar logic, we could probably reuse this method over there. Will raise a separate issue for that though.